packages/playwright-core/src/server/webkit/wkPage.ts

```diff
@@ -1029,13 +1029,13 @@ export class WKPage implements PageDelegate {
     // TODO(einbinder) this will fail if we are an XHR document request
     const isNavigationRequest = event.type === 'Document';
     const documentId = isNavigationRequest ? event.loaderId : undefined;
-    let route = null;
+    const request = new WKInterceptableRequest(session, frame, event, redirectedFrom, documentId);
     // We do not support intercepting redirects.
     if (this._page.needsRequestInterception() && !redirectedFrom)
-      route = new WKRouteImpl(session, event.requestId);
-    const request = new WKInterceptableRequest(session, route, frame, event, redirectedFrom, documentId);
+      request._willBeIntercepted = true;
```

Comment:
Let's have a map from request id to `Protocol.Network.requestWillBeSentPayload` and only create `WKInterceptableRequest` later, once the interception event arrives.